core/data/src/test/java/uk/gov/gchq/gaffer/data/element/function/TypeValueToTupleTest.java

```diff
@@ -0,0 +1,132 @@
+import uk.gov.gchq.gaffer.exception.SerialisationException;
+import uk.gov.gchq.gaffer.jsonserialisation.JSONSerialiser;
+import uk.gov.gchq.gaffer.types.TypeValue;
+import uk.gov.gchq.koryphe.function.FunctionComposite;
+import uk.gov.gchq.koryphe.function.FunctionTest;
+import uk.gov.gchq.koryphe.impl.function.Length;
+import uk.gov.gchq.koryphe.impl.function.ToString;
+import uk.gov.gchq.koryphe.tuple.Tuple;
+import uk.gov.gchq.koryphe.tuple.function.TupleAdaptedFunctionComposite;
+
+import java.util.Arrays;
+import java.util.function.Function;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+public class TypeValueToTupleTest extends FunctionTest<TypeValueToTuple> {
+
+    @Test
+    public void shouldHandleNullInput() {
+        // Given
+        final TypeValueToTuple function = getInstance();
+
+        // When
+        final Tuple<String> result = function.apply(null);
+
+        // Then
+        assertNull(result);
+    }
+
+    @Test
+    public void shouldConvertTypeValueToTuple() {
+        // Given
+        final TypeValue typeValue = new TypeValue("type", "value");
+        final TypeValueToTuple function = getInstance();
+
+        // When
+        final Tuple<String> result = function.apply(typeValue);
+
+        // Then
+        assertEquals("type", result.get("type"));
+        assertEquals("value", result.get("value"));
+    }
+
+    @Test
+    public void shouldGetAndSetUsingCompositeFunction() {
+        // Given
+        final TypeValue typeValue = new TypeValue("type", "value");
+        final Function<Object, Object> compositeFunction = new FunctionComposite(Lists.newArrayList(
+                new TypeValueToTuple(),
+                new TupleAdaptedFunctionComposite.Builder()
+                        .select(new String[]{"value"})
+                        .execute(new FunctionComposite(Arrays.asList(
+                                new Length(),
+                                new ToString()
+                        )))
+                        .project(new String[]{"type"})
+                        .build()
+        ));
+
+        // When
+        compositeFunction.apply(typeValue);
+
+        // Then
+        assertEquals(new TypeValue("5", "value"), typeValue);
+    }
+
+    @Override
+    protected TypeValueToTuple getInstance() {
+        return new TypeValueToTuple();
+    }
+
+    @Override
+    protected Iterable getDifferentInstancesOrNull() {
+        return null;
+    }
+
+    @Override
+    protected Class<? extends TypeValueToTuple> getFunctionClass() {
+        return TypeValueToTuple.class;
+    }
+
+    @Override
+    protected Class[] getExpectedSignatureInputClasses() {
+        return new Class[]{TypeValue.class};
+    }
+
+    @Override
+    protected Class[] getExpectedSignatureOutputClasses() {
+        return new Class[]{TypeValueTuple.class};
```

Comment:
Perhaps this should be changed back to `Class[]{String.class}`. Despite the slightly confusing layout, checking that the `Tuple` contains the correct type is potentially more useful, as explained in https://github.com/gchq/koryphe/issues/225#issuecomment-896004455.

Reply:
After some more understanding, it is okay for Gaffer to implement its own concrete implementations of a `Tuple` and test for those types (see https://github.com/gchq/koryphe/issues/225#issuecomment-896103601). No changes should be required here.